### ISSUE TITLE
Support reading SparseVectors and Vectors 

### ISSUE THREAD

**Issue by abditag2** [uber/petastorm#425]
Would be very useful if petastorm can read these two data types from HDFS.

**Comment by selitvin** [comment]
Agreed. We can try and add. Not sure about the time-frame for this though.

**Comment by mengxr** [comment]
Efficient conversion requires Scala UDFs. Maybe we should add utility methods to Spark so in petastorm we can do the following:

~~~python
spark._jvm.ml.util.registerVectorToArrayUDF("toArray")
df.selectExpr("toArray(features) as features")
~~~

This approach doesn't require Scala code in petastorm. Let me create a Spark JIRA. cc: @WeichenXu123

**Comment by mengxr** [comment]
FYI. The UDF was merged into Spark master: https://github.com/apache/spark/pull/26910
